### ISSUE TITLE
Unable to terminate KSQL query with a - (hyphen, dash) in the query id

### ISSUE THREAD

**Issue by DW-Pete** [confluentinc/ksql#1888]
I created a CSAS stream like:
```
ksql> CREATE STREAM `O-WMS.ITEM_BARCODE_JSON` WITH (VALUE_FORMAT='JSON') AS SELECT \
ITM.C1, UPC.C2 FROM  ITM \
> INNER JOIN  UPC ON \
> (ITM.ITEM_ID = UPC.ITEM_ID);
 Message                    
----------------------------
 Stream created and running 
----------------------------
```
Note back quotes on stream name.  I realised that I should have used a different name for the stream.  I tried to terminate the stream using these:
```
ksql> terminate `CSAS_O-WMS.ITEM_BARCODE_JSON_62`;
java.lang.Exception: No running query with id `CSAS_O-WMS.ITEM_BARCODE_JSON_62` was found
ksql> terminate CSAS_O-WMS.ITEM_BARCODE_JSON_62;
line 1:17: mismatched input '-' expecting ';'
Caused by: org.antlr.v4.runtime.InputMismatchException

ksql> terminate 'CSAS_O-WMS.ITEM_BARCODE_JSON_62';

line 1:11: mismatched input ''CSAS_O-WMS.ITEM_BARCODE_JSON_62'' expecting {'ADD', 'APPROXIMATE', 'AT', 'CONFIDENCE', 'NO', 'SUBSTRING', 'POSITION', 'TINYINT', 'SMALLINT', 'INTEGER', 'DATE', 'TIME', 'TIMESTAMP', 'INTERVAL', 'YEAR', 'MONTH', 'DAY', 'HOUR', 'MINUTE', 'SECOND', 'ZONE', 'OVER', 'PARTITION', 'RANGE', 'ROWS', 'PRECEDING', 'FOLLOWING', 'CURRENT', 'ROW', 'STRUCT', 'VIEW', 'REPLACE', 'GRANT', 'REVOKE', 'PRIVILEGES', 'PUBLIC', 'OPTION', 'EXPLAIN', 'ANALYZE', 'FORMAT', 'TYPE', 'TEXT', 'GRAPHVIZ', 'LOGICAL', 'DISTRIBUTED', 'TRY', 'SHOW', 'TABLES', 'QUERY', 'SCHEMAS', 'CATALOGS', 'COLUMNS', 'COLUMN', 'USE', 'PARTITIONS', 'FUNCTIONS', 'FUNCTION', 'TO', 'SYSTEM', 'BERNOULLI', 'POISSONIZED', 'TABLESAMPLE', 'RESCALED', 'ARRAY', 'MAP', 'SET', 'RESET', 'SESSION', 'DATA', 'START', 'TRANSACTION', 'COMMIT', 'ROLLBACK', 'WORK', 'ISOLATION', 'LEVEL', 'SERIALIZABLE', 'REPEATABLE', 'COMMITTED', 'UNCOMMITTED', 'READ', 'WRITE', 'ONLY', 'CALL', 'NFD', 'NFC', 'NFKD', 'NFKC', 'IF', 'NULLIF', 'COALESCE', IDENTIFIER, DIGIT_IDENTIFIER, QUOTED_IDENTIFIER, BACKQUOTED_IDENTIFIER}
Caused by: org.antlr.v4.runtime.InputMismatchException
```

I need to find a way to terminate the query so I can drop the stream

**Comment by saubury-iag** [comment]
I just hit the exact same problem.  I have a misnamed stream (with a hyphen) and cannot terminate the query to remove it

**Comment by apurvam** [comment]
Seems to be a dup of #1466

**Comment by DW-Pete** [comment]
> Seems to be a dup of #1466

However a fix for 1466 could be just a documentation change (to say not supported). This one is specifically for a query forced to have a hyphen in the name (and actually running!) not working with TERMINATE.

**Comment by big-andy-coates** [comment]
Yeah #1466 is an issue with a hyphen in a field name - this is a hyphen in the stream name. Related, but not the same issue / fix.

**Comment by agavra** [comment]
@big-andy-coates I think this is a dup of #1466 - they are both about hyphens in stream/table names (hyphens in fields has been fixed as part of the quoted identifier work). Similarly, the fix for this is the same as #1466. If we prohibit streams/tables with hyphens, then there will not be queries with hyphens in their names so we won't run into this.

I will close this out and track the fix on #1466, but feel free to re-open if I misunderstood.
